examples/java/spring-boot/src/main/java/uk/gov/api/springboot/infrastructure/ScheduleHandler.java

```diff
@@ -0,0 +1,37 @@
+package uk.gov.api.springboot.infrastructure;
+
+import java.util.Collection;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import uk.gov.api.springboot.domain.model.Registry;
+import uk.gov.api.springboot.domain.model.repositories.ApiStorage;
+import uk.gov.api.springboot.domain.services.FetcherService;
+
+@Component
+public class ScheduleHandler {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ScheduleHandler.class);
+  private final FetcherService fetcherService;
+  private final Registry registry;
+  private final ApiStorage storage;
+
+  public ScheduleHandler(FetcherService fetcherService, Registry registry, ApiStorage storage) {
+    this.fetcherService = fetcherService;
+    this.registry = registry;
+    this.storage = storage;
+  }
+
+  @Scheduled(fixedRateString = "${apis-fetch-rate.in.milliseconds}")
+  public void fetchAndSaveApis() {
+    LOGGER.info("Fetching and saving APIS");
+
+    var allApis =
+        registry.findAll().stream().map(fetcherService::fetch).flatMap(Collection::stream).toList();
+
+    storage.save(allApis);
```

Comment:
we could alternatively make `storage.save()` take a single `Api`, then we can start saving `Api`s as soon as they're available, rather than wait for _all_ the fetching to finish?

Reply:
something like this? https://github.com/co-cddo/federated-api-model/pull/102/commits/964496357ba581bbacc805982a706cb4a31895fb